src/emulator/modules/SceTouch/SceTouch.h

```diff
@@ -17,8 +17,12 @@
 
 #pragma once
 
+#include <SDL.h>
 #include <module/module.h>
 
+int handle_touch_event(SDL_TouchFingerEvent finger);
```

Comment:
Try to pass the parameter as reference. and foward declare it. That way SDL include is unneccessary

```suggestion
struct SDL_TouchFingerEvent;
typedef SDL_TouchFingerEvent SDL_TouchFingerEvent;

int handle_touch_event(SDL_TouchFingerEvent &finger);
```

Same with source. This will save some times btw (if you dont pass it as reference it will copy the data)

Reply:
i don't quite understand this, i get the passing by reference, but not what you did with the structs.
Should i just copy it?

Reply:
That's foward declaring, basically you declare the existence of a struct, than filling the struct info later somewhere (in the cpp file for example). If you can foward declare it, you can remove the SDL.h include. You should avoid unneccessary inclusion when you can.

 Sure i still suggest passing by reference

Reply:
And sure, try to copy it, and remove including SDL.h in SceTouch.h if it fails, than we will only pass by reference :p

Reply:
by just copy it i meant your suggested change